docs/api/custom-integration.md

```diff
@@ -0,0 +1,115 @@
+
+```kotlin title="Specifying the editorConfigDefaults using an '.editorconfig' file"
+val ktLintRuleEngine =
+    KtLintRuleEngine(
+        ruleProviders = KTLINT_API_CONSUMER_RULE_PROVIDERS,
+        editorConfigDefaults = EditorConfigDefaults.load(
+            path = Paths.get("/some/path/to/editorconfig/file/or/directory"),
+            propertyTypes = KTLINT_API_CONSUMER_RULE_PROVIDERS.propertyTypes(),
+        )
+  )
+```
+
+The `EditorConfigDefaults` property can also be specified programmatically as is shown below:
+
+```kotlin title="Specifying the editorConfigDefaults programmatically"
+val ktLintRuleEngine =
+    KtLintRuleEngine(
+        ruleProviders = KTLINT_API_CONSUMER_RULE_PROVIDERS,
+        editorConfigDefaults = EditorConfigDefaults(
+            org.ec4j.core.model.EditorConfig
+                .builder()
+                // .. add relevant properties
+                .build()
+        )
+    )
+```
+
+### Lint & format
+
+Once the `KtLintRuleEngine` has been defined, it is ready to be invoked for each file or code snippet that has to be linted or formatted. The the `lint` and `format` functions take a `Code` instance as parameter. Such an instance can either be created from a file
+```kotlin title="Code from file"
+val code = Code.fromFile(
+    File("/some/path/to/file")
+)
+```
+or a code snippet (set `script` to `true` to handle the snippet as Kotlin script):
+```kotlin title="Code from snippet"
+val code = Code.fromSnippet(
+    """
+    val code = "some-code"
+    """.trimIndent()
+)
+```
+
+The `lint` function is invoked with a lambda which is called each time a `LintError` is found and does not return a result.
+```kotlin title="Specifying the editorConfigDefaults programmatically"
+ktLintRuleEngine
+  .lint(codeFile) { lintError ->
+    // handle
+  }
+```
+
+The `format` function is invoked with a lambda which is called each time a `LintError` is found and returns the formatted code as result. Note that the `LintError` should be inspected for errors that could not be autocorrected.
+```kotlin title="Specifying the editorConfigDefaults programmatically"
+val formattedCode =
+    ktLintRuleEngine
+        .format(codeFile) { lintError ->
+          // handle
+        }
+```
+
+## Logging
+
+Ktlint uses the `io.github.microutils:kotlin-logging` which is a `slf4j` wrapper. As API consumer you can choose which logging framework you want to use and configure that framework to your exact needs. The [basic API Consumer](https://github.com/pinterest/ktlint/blob/master/ktlint-api-consumer/src/main/kotlin/com/example/ktlint/api/consumer/KtlintApiConsumer.kt) contains an example with `org.slf4j:slf4j-simple` as logging provider and a customized configuration which shows logging at `DEBUG` level for all classes except one specific class which only displays logging at `WARN` level.
```

Comment:
So, one thing that is not clear to me is: if I use `ktlint-api-consumer` and and I also use `kotlin-logging` for my logs, am I going to see logs from ktlint when I use `ktLintRuleEngine.format()`? If yes, how can I completely turn off every log from KtLint only (so, without this affecting the usages of `kotlin-logging` outside KtLint)?

Reply:
I have added an additional example in the logging configuration that explains how to disable all logging of ktlint only.

Reply:
Thanks!